error: parameter of opaque type 'Gdk.GLContext' not allowed

source code:
```zig
const std = @import("std");
const gtk = @import("gtk");
const core = gtk.core;
const Application = gtk.Application;
const ApplicationWindow = gtk.ApplicationWindow;
const Box = gtk.Box;
const Button = gtk.Button;
const Widget = gtk.Widget;
const GLArea = gtk.GLArea;
const Window = gtk.Window;
const GApplication = core.Application;
const gdk = gtk.Gdk;
const GLContext = gdk.GLContext;

pub fn printHello() void {
    std.log.info("Hello World", .{});
}

fn on_realize() void {
    std.log.debug("realized", .{});
}

fn on_render(glarea: *GLArea, context: GLContext) bool {
    _ = glarea;
    _ = context;

    std.log.debug("render", .{});

    return true;
}

pub fn activate(arg_app: *GApplication) void {
    const app = arg_app.tryInto(Application).?;
    var appwindow = ApplicationWindow.new(app);
    var window = appwindow.into(gtk.Window);
    window.setTitle("hello world");
    window.setDefaultSize(400, 300);

    var frame = gtk.Frame.new("window");
    var glarea = GLArea.new();

    _ = glarea.connectRender(on_render, .{}, .{});
    // _ = glarea.connect("render", on_render, .{}, .{}, &[_]type{bool, *GLArea, *GLContext});

    frame.setChild(glarea.into(Widget));
    window.setChild(frame.into(Widget));

    window.present();
}

pub fn main() u8 {
    var app = Application.new("org.gtk.example", .{});
    defer app.__call("unref", .{});
    _ = app.connect("activate", activate, .{}, .{}, &.{void, *GApplication});
    return @intCast(app.__call("run", .{std.os.argv}));
}

```
I get the following error:
```
run
└─ run example
   └─ zig build-exe example Debug native 1 errors
main.zig:23:31: error: parameter of opaque type 'Gdk.GLContext' not allowed
fn on_render(glarea: *GLArea, context: GLContext) bool {
                              ^~~~~~~~~~~~~~~~~~
/home/hasan/.cache/zig/p/12201030b8083fb96a0a5b077217dc87f779018244d569974a287ad43c1ed7a8ea43/Gdk.zig:2318:23: note: opaque declared here
pub const GLContext = opaque {
                      ^~~~~~
referenced by:
    activate: main.zig:42:30
    main: main.zig:54:33
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
error: the following command failed with 1 compilation errors:
/snap/zig/11356/zig build-exe -I/usr/include/gtk-4.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/x86_64-linux-gnu -I/usr/include/webp -I/usr/include/graphene-1.0 -I/usr/lib/x86_64-linux-gnu/graphene-1.0/include -lgtk-4 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -lgdk_pixbuf-2.0 -lcairo-gobject -lcairo -lvulkan -lgraphene-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/fribidi -I/usr/include/x86_64-linux-gnu -I/usr/include/libxml2 -I/usr/include/lua5.2 -I/usr/include/SDL2 -I/usr/include/uchardet -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -D_REENTRANT -I/usr/include/libdrm -I/usr/include/sixel -I/usr/include/spirv_cross -lmpv -ODebug --dep gtk -Mroot=/home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/main.zig -Mgtk=/home/hasan/.cache/zig/p/12201030b8083fb96a0a5b077217dc87f779018244d569974a287ad43c1ed7a8ea43/Gtk.zig -lc --cache-dir /home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/zig-cache --global-cache-dir /home/hasan/.cache/zig --name example --listen=- 
Build Summary: 0/5 steps succeeded; 1 failed (disable with --summary none)
run transitive failure
└─ run example transitive failure
   ├─ zig build-exe example Debug native 1 errors
   └─ install transitive failure
      └─ install example transitive failure
         └─ zig build-exe example Debug native (reused)
error: the following build command failed with exit code 1:
/home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/zig-cache/o/124186cad66f582f698e34d0af0c5eb2/build /snap/zig/11356/zig /home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4 /home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/zig-cache /home/hasan/.cache/zig --seed 0xd56ced2a -Z5c8706d432351596 run
```
When I change the `on_render` function signature to `fn on_render(glarea: *GLArea, context: *GLContext) bool`, I get the following error:
```
run
└─ run example
   └─ install
      └─ install example
         └─ zig build-exe example Debug native 1 errors
/home/hasan/.cache/zig/p/12201030b8083fb96a0a5b077217dc87f779018244d569974a287ad43c1ed7a8ea43/core.zig:475:47: error: parameter of opaque type 'Gdk.GLContext' not allowed
            pub fn invoke(arg1: signature[1], arg2: signature[2], self: *Self) callconv(.C) signature[0] {
                                              ^~~~~~~~~~~~~~~~~~
/home/hasan/.cache/zig/p/12201030b8083fb96a0a5b077217dc87f779018244d569974a287ad43c1ed7a8ea43/Gdk.zig:2318:23: note: opaque declared here
pub const GLContext = opaque {
                      ^~~~~~
referenced by:
    c_closure: /home/hasan/.cache/zig/p/12201030b8083fb96a0a5b077217dc87f779018244d569974a287ad43c1ed7a8ea43/core.zig:584:56
    connect__anon_13981: /home/hasan/.cache/zig/p/12201030b8083fb96a0a5b077217dc87f779018244d569974a287ad43c1ed7a8ea43/core.zig:383:61
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
error: the following command failed with 1 compilation errors:
/snap/zig/11356/zig build-exe -I/usr/include/gtk-4.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/x86_64-linux-gnu -I/usr/include/webp -I/usr/include/graphene-1.0 -I/usr/lib/x86_64-linux-gnu/graphene-1.0/include -lgtk-4 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -lgdk_pixbuf-2.0 -lcairo-gobject -lcairo -lvulkan -lgraphene-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/fribidi -I/usr/include/x86_64-linux-gnu -I/usr/include/libxml2 -I/usr/include/lua5.2 -I/usr/include/SDL2 -I/usr/include/uchardet -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -D_REENTRANT -I/usr/include/libdrm -I/usr/include/sixel -I/usr/include/spirv_cross -lmpv -ODebug --dep gtk -Mroot=/home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/main.zig -Mgtk=/home/hasan/.cache/zig/p/12201030b8083fb96a0a5b077217dc87f779018244d569974a287ad43c1ed7a8ea43/Gtk.zig -lc --cache-dir /home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/zig-cache --global-cache-dir /home/hasan/.cache/zig --name example --listen=- 
Build Summary: 0/5 steps succeeded; 1 failed (disable with --summary none)
run transitive failure
└─ run example transitive failure
   ├─ zig build-exe example Debug native 1 errors
   └─ install transitive failure
      └─ install example transitive failure
         └─ zig build-exe example Debug native (reused)
error: the following build command failed with exit code 1:
/home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/zig-cache/o/124186cad66f582f698e34d0af0c5eb2/build /snap/zig/11356/zig /home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4 /home/hasan/Documents/coding/zig/projs/mpv-projs/zmpv-gtk4/zig-cache /home/hasan/.cache/zig --seed 0x36b86e47 -Z4c425a3c372a925b run

```

--- a/src/gir.zig
+++ b/src/gir.zig
@@ -1016,7 +1016,15 @@ pub const SignalInfo = struct {
         try writer.print("return self.connect(\"{s}\", handler, args, flags, &[_]type{{", .{raw_name});
         const return_type = self.asCallable().returnType();
         defer return_type.asBase().deinit();
-        if (self.asCallable().mayReturnNull()) {
+        var interface_returned = false;
+        if (return_type.interface()) |child_type| {
+            defer child_type.deinit();
+            switch (child_type.type()) {
+                .Enum, .Flags => {},
+                else => interface_returned = true,
+            }
+        }
+        if (self.asCallable().mayReturnNull() or interface_returned) {
             try writer.print("{&*}", .{return_type});
         } else {
             try writer.print("{&}", .{return_type});
@@ -1024,7 +1032,7 @@ pub const SignalInfo = struct {
         try writer.print(", *{s}", .{container_name});
         var iter = self.asCallable().argsIter();
         while (iter.next()) |arg| {
-            try writer.print(", {$}", .{arg});
+            try writer.print(", {$*}", .{arg});
         }
         try writer.writeAll("});\n");
         try writer.writeAll("}\n");
@@ -2093,9 +2101,13 @@ pub const ArgInfo = struct {
     pub fn format(self: ArgInfo, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
         _ = options;
         var option_type_only = false;
+        var option_signal_param = false;
         for (fmt) |ch| {
             if (ch == '$') {
                 option_type_only = true;
+            }
+            if (ch == '*') {
+                option_signal_param = true;
             }
         }
         if (!option_type_only) {
@@ -2104,7 +2116,18 @@ pub const ArgInfo = struct {
         }
         const arg_type = self.type();
         defer arg_type.asBase().deinit();
-        if (self.direction() != .In) {
+        if (option_signal_param) {
+            if (arg_type.interface()) |child_type| {
+                defer child_type.deinit();
+                switch (child_type.type()) {
+                    .Enum, .Flags => option_signal_param = false,
+                    else => {},
+                }
+            } else {
+                option_signal_param = false;
+            }
+        }
+        if (self.direction() != .In or option_signal_param) {
             if (self.isOptional()) {
                 if (self.mayBeNull()) {
                     try writer.print("{_&?*}", .{arg_type});

